src/sentry/monitors/models.py

```diff
@@ -474,13 +508,28 @@ class MonitorEnvironment(Model):
     monitor = FlexibleForeignKey("sentry.Monitor")
     environment = FlexibleForeignKey("sentry.Environment")
     status = BoundedPositiveIntegerField(
-        default=MonitorStatus.ACTIVE, choices=MonitorStatus.as_choices()
+        default=MonitorStatus.ACTIVE,
+        choices=MonitorStatus.as_choices(),
     )
-    next_checkin = models.DateTimeField(null=True)  # the expected time of the next check-in
-    next_checkin_latest = models.DateTimeField(
-        null=True
-    )  # the latest expected time of the next check-in (includes check-in margin)
+    """
+    The MonitorStatus of the monitor. This is denormalized from the check-ins
+    list, since it would be possible to determine this by looking at recent
+    check-ins. It is denormalized for simplicity.
+    """
+    next_checkin = models.DateTimeField(null=True)
+    """
+    The expected time that the next-checkin will occur
+    """
+    next_checkin_latest = models.DateTimeField(null=True)
+    """
+    The latest expected time that the next-checkin will occur. This is computed
+    using the user-configured margin for the monitor.
+    """
     last_checkin = models.DateTimeField(null=True)
+    """
+    Time of the most recent user-check in. This does not include auto-generated
```

Comment:
```suggestion
    Time of the start of the most recent user-check in. This does not include auto-generated
```

Reply:
actually is it EXPECTED time of the most recent check in?

Reply:
no we update it on every `mark_failed` or `mark_ok`

Reply:
```suggestion
    date_added time of the most recent user-check in. This does not include auto-generated
```

How about this?

Reply:
yeah that works